### PR TITLE
add new function and event

### PR DESCRIPTION
**Pull request by PeterGamez**

### CODE REVIEW

File: src/structures/LavalinkManager.ts [hunk 1]
```diff
@@ -539,6 +542,18 @@ export class LavalinkManager extends EventEmitter {
...
+                    this.emit("playerMute", player, true);
+                } else if (update.mute === false) {
+                    this.emit("playerMute", player, false);
+                }
+
+                if (update.deaf === true) {
+                    this.emit("playerDeaf", player, true);
+                } else if (update.deaf === false) {
+                    this.emit("playerDeaf", player, false);
+                }
```
Comment (Tomato6966) [inline src/structures/LavalinkManager.ts:556]: Good idea of new events
Shouldn't you also be able to track for serverDeaf too?(i think it's server_deaf)

Then the event chould have 3 props: player, selfDeaf, serverDeaf

Comment (PeterGamez) [reply src/structures/LavalinkManager.ts:556]: yes it serverDeaf. i see on some bot are auto server deaf (self) when any someone are un deaf

**Review by Tomato6966 — COMMENTED** [review]
Modified the voice state handling event emittions, and added new emptyVC handler, testing will be done soon

**Review by Tomato6966 — COMMENTED** [review]
Merged, Resolved conflicts, and added new stuff

**Review by Tomato6966 — APPROVED** [review]

**Comment by Tomato6966** [comment]
- Added new player events:
    - **`playerMuteChange`**** ➡️ **`(player, selfMuted, serverMuted) => {}`
    *Triggered when the player's voice state related to muting changed*

    - **`playerDeafChange`** ➡️ `(player, selfDeafed, serverDeafed) => {}`
    *Triggered when the player's voice state related to deafing changed*

    - **`playerSuppressChange`** ➡️ `(player, suppress) => {}`
    *Triggered when the player's voice state related to suppressing changed*

    - **`playerQueueEmptyStart`** ➡️ `(player, timeoutMs) => {}`
    *Triggered when the queue empty handler started (the timeout)*

    - **`playerQueueEmptyEnd`** ➡️ `(player) => {}`
    *Triggered when the queue empty handler finished (successfully) and thus destroyed the player*

    - **`playerQueueEmptyCancel`** ➡️ `(player) => {}`
    *Triggered when the queue empty handler cancelled (e.g. because a new track got added)*

    - **`playerVoiceEmptyStart`** ➡️ `(player, timeoutMs) => {}`
    *Triggered when the voice empty handler started (the timeout)*

    - **`playerVoiceEmptyEnd`** ➡️ `(player) => {}`
    *Triggered when the voice empty handler finished (successfully) and thus destroyed the player*

    - **`playerVoiceEmptyCancel`** ➡️ `(player, userId) => {}`
    *Triggered when the voice empty handler cancelled (e.g. when a user rejoined)*
- Added new managerOption: `playerOption.onEmptyPlayerVoice.destroyAfterMs`
  - When provided and bigger than 0, then the player get's destroyed after that ms when the voice get's empty
  - Added voice state handling events for easier integration (suggested by @PeterGamez)

- Added the new events and configuration to the docs

**Pull request merged**